libpod/runtime_ctr.go

```diff
@@ -915,6 +915,44 @@ func (r *Runtime) evictContainer(ctx context.Context, idOrName string, removeVol
 	return id, cleanupErr
 }
 
+// DependMap populates a map of all containers depending on the specified container
+func (r *Runtime) DependMap(ctx context.Context, ctr *Container, ctrMap map[string]*Container) error {
+	r.lock.RLock()
+	defer r.lock.RUnlock()
+	// Grab all containers
+	allctrs, err := r.GetAllContainers()
+	if err != nil {
+		return err
+	}
+	// Make sure current Container is mapped
+	if ctrMap[ctr.ID()] == nil {
+		ctrMap[ctr.ID()] = ctr
+	}
+	return r.dependMap(ctx, ctr, allctrs, ctrMap)
+}
+
+// dependMap populates of map of all containers depending on the specified container
+func (r *Runtime) dependMap(ctx context.Context, searchCtr *Container, allCtrs []*Container, ctrMap map[string]*Container) error {
+	for _, ctr := range allCtrs {
+		if ctr == searchCtr {
+			continue
+		}
+		if ctrMap[ctr.ID()] != nil {
+			continue
+		}
+		for _, cid := range ctr.Dependencies() {
```

Comment:
if a container depends on infra within a pod, will this remove infra but keep the pod in existence? Could bring up some weird issues.

Reply:
I think it'll just hard-fail because infra containers can't be removed except by removing their pod - so `podman rm -f --depend infra_ctr` should fail without doing anything

Reply:
./bin/podman rm --depend 35965a814e0b
Error: container 35965a814e0ba301bc5d19bd1acaa1809af443a09c4d3f9ebacc02a454bc72f3 is the infra container of pod f72b7a1abf8f76a69e4a859542103db4184bcd7d07b7afaf73608eb6db870a0e and cannot be removed without removing the pod